tests/testsupport/broker.py

```diff
@@ -0,0 +1,62 @@
+import socket
+
+import pytest
+
+
+class FakeBroker:
+    def __init__(self):
+        # Bind to "localhost" for maximum performance, as described in:
+        # http://docs.python.org/howto/sockets.html#ipc
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+        sock.settimeout(30)
+        sock.bind(("localhost", 1888))
+        sock.listen(1)
+
+        self._sock = sock
+        self._conn = None
+
+    def start(self):
+        if self._sock is None:
+            raise ValueError('Socket is not open')
+
+        (conn, address) = self._sock.accept()
+        conn.settimeout(10)
+        self._conn = conn
+
+    def finish(self):
+        if self._conn is not None:
+            self._conn.close()
+            self._conn = None
+
+        if self._sock is not None:
+            self._sock.close()
+            self._sock = None
+
+    def expect_packet(self, packet_expected):
+        if self._conn is None:
+            raise ValueError('Connection is not open')
+
+        packet_in = self._conn.recv(len(packet_expected))
+        if not packet_in:
+            raise RuntimeError('Connection was closed')
+        assert packet_in == packet_expected
```

Comment:
I had expected pytest rewrite the assertion: https://docs.pytest.org/en/latest/assert.html. I think that the issue is that the assertion is made in another module, rather than the test module, although pytest is imported there too.

I've re-written the test as follows:

```
            connect_packet = paho_test.gen_connect(
                "01-con-discon-success-fail", keepalive=60,  # Wrong client id
                proto_name="MQIsdp", proto_ver=3)
            packet_in = fake_broker.receive_packet(len(connect_packet))
            assert packet_in == connect_packet
```

And pytest now says:

```
>           assert packet_in == connect_packet
E           AssertionError: assert '\x10#\x00\x0...iscon-success' == '\x10(\x00\x06...-success-fail'
E             - \x10#\x00\x06MQIsdp\x03\x02\x00<\x00\x1501-con-discon-success
E             ?     ^                                  ^
E             + \x10(\x00\x06MQIsdp\x03\x02\x00<\x00\x1a01-con-discon-success-fail
E             ?     ^                                  ^                     +++++
```

I hope you'll agreed that this is much more useful than a simple `print`.